infrastructure/modules/migrations/main.tf

```diff
@@ -0,0 +1,108 @@
+  data "aws_secretsmanager_secret_version" "etl_db" {
+      secret_id = var.etl_db.db_instance_master_user_secret_arn
+  }
+  data "aws_secretsmanager_secret_version" "fhir_db" {
+      secret_id = var.fhir_db.db_instance_master_user_secret_arn
+  }
+
+  locals {
+    etl_db_password  = data.aws_secretsmanager_secret_version.etl_db.secret_string
+    fhir_db_password = data.aws_secretsmanager_secret_version.fhir_db.secret_string
+    account_name     = "${var.region}-${var.tier}"
+    multi_az         = var.multi_az
+    table_mappings   = jsonencode({
+      "rules": [
+          {
+              "rule-id": 1,
+              "rule-name": "etl-selection-rule",
+              "rule-type": "selection",
+              "rule-action": "include",
+              "object-locator": {
+                  "schema-name": "npd",
+                  "table-name": "%"
+              },
+              "filters": []
+          },
+          {
+              "rule-id": 2,
+              "rule-name": "etl-transformation-rule",
+              "rule-type": "transformation",
+              "rule-target": "table",
+              "rule-action": "rename",
+              "object-locator": {
+                  "schema-name": "npd",
+                  "table-name": "%"
+              },
+            "value": "${module.database_migration_service.endpoints["fhir-api-destination"].database_name}-migrated" # naming scheme suggestion
+          }
+      ]
+    })
+  }
+
+  module "networking" {
+  source = "../../modules/networking"
+
+  vpc_id       = var.networking.vpc_id
+  account_name = local.account_name
+}
+
+module "database_migration_service" {
+  source  = "terraform-aws-modules/dms/aws"
+  version = "~> 2.0"
+
+  # Subnet group
+  repl_subnet_group_name        = module.networking.private_subnet_group_name
+  repl_subnet_group_description = "The DMS ETL replication subnet group"
+  repl_subnet_group_subnet_ids  = module.networking.private_subnet_ids
+
+  endpoints = {
+    etl-source = {
+      database_name = "npd_etl" # possible to reference this by env?
+      endpoint_id   = "npd-east-${var.tier}-etl-source"
```

Comment:
${var.account_name} is a combination of `npd-east-` and `${var.tier}`, so you can use that instead of piping in tier directly.